aws/resource_aws_route53_health_check.go

```diff
@@ -341,7 +346,7 @@ func resourceAwsRoute53HealthCheckCreate(d *schema.ResourceData, meta interface{
 
 	d.SetId(aws.StringValue(resp.HealthCheck.Id))
 
-	if err := keyvaluetags.Route53UpdateTags(conn, d.Id(), route53.TagResourceTypeHealthcheck, map[string]interface{}{}, d.Get("tags").(map[string]interface{})); err != nil {
+	if err := keyvaluetags.Route53UpdateTags(conn, d.Id(), route53.TagResourceTypeHealthcheck, nil, tags); err != nil {
```

Comment:
i _think_ here we'll need to pass in `tags.Map()` instead of `tags` since the `Route53UpdateTags` method will call `New` on the tags and result in a new `KeyValueTags` instance but without the values originally in it

Reply:
Submitted https://github.com/hashicorp/terraform-provider-aws/pull/19011 to cover this nonintuitive situation 👍